client.go

```diff
@@ -526,7 +537,16 @@ func (c *Client) send(r *Request) (*amqp.Delivery, error) {
 	timeoutChan := r.startTimeout()
 
 	c.debugLog("client: queuing request %s", r.Publishing.CorrelationId)
-	c.requests <- r
+
+	select {
+	case c.requests <- r:
+		// successful send.
+	case <-timeoutChan:
```

Comment:
This also makes me realise that if something before this request is taking time, the timeout from this message is used. I guess that's OK since you probably mean "tell me if I don't have a response within X" but maybe there should be a way to separate network timeout vs. total timeout? Maybe just something for the future.

Reply:
Yea I've also thought about that. But I'm thinking that "I want a reply within 1 second" should not have `Send()` block for 2 seconds.